aws_lambda_powertools/utilities/parameters/secrets.py

```diff
@@ -115,6 +120,65 @@ def _get_multiple(self, path: str, **sdk_options) -> Dict[str, str]:
         """
         raise NotImplementedError()
 
+    def _set(
+        self,
+        name: str,
+        value: Union[str, dict, bytes],
+        *,  # force keyword arguments
+        client_request_token: Optional[str] = None,
+        version_stages: Optional[list[str]] = None,
+        **sdk_options,
+    ) -> str:
+        """
+        Modifies the details of a secret, including metadata and the secret value.
+
+        Parameters
+        ----------
+        name: str
+            The ARN or name of the secret to add a new version to.
+        value: str or bytes
+            Specifies text data that you want to encrypt and store in this new version of the secret.
+        client_request_token: str, optional
+            This value helps ensure idempotency. Recommended that you generate
+            a UUID-type value to ensure uniqueness within the specified secret.
+            This value becomes the VersionId of the new version. This field is
+            autopopulated if not provided.
+        version_stages: list[str], optional
+            Specifies a list of staging labels that are attached to this version of the secret.
+        sdk_options: dict, optional
+            Dictionary of options that will be passed to the Secrets Manager update_secret API call
+
+        Returns:
+        -------
+            Version ID of the newly created version of the secret.
+
+        URLs:
+        -------
+            https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/secretsmanager/client/put_secret_value.html
+        """
+
+        sdk_options["SecretId"] = name
+
+        if isinstance(value, dict):
+            value = json.dumps(value)
+
+        if isinstance(value, bytes):
+            sdk_options["SecretBinary"] = value
+        else:
+            sdk_options["SecretString"] = value
+
+        if version_stages:
+            sdk_options["VersionStages"] = version_stages
+        if client_request_token:
+            sdk_options["ClientRequestToken"] = client_request_token
+
+        try:
+            value = self.client.put_secret_value(**sdk_options)
+            return value["VersionId"]
+        except ClientError as exc:
+            if exc.response["Error"]["Code"] != "ResourceNotFoundException":
+                raise SetParameterError(str(exc)) from exc
```

Comment:
we need a more specific secret exception here

Reply:
Done